test/Sign.SignatureProviders.KeyVault.Test/KeyVaultServiceProviderTests.cs

```diff
@@ -77,14 +78,14 @@ public void GetSignatureAlgorithmProvider_ReturnsSameInstance()
         {
             KeyVaultServiceProvider provider = new(tokenCredential, KeyVaultUrl, CertificateName);
 
-            List<ISignatureAlgorithmProvider> signatureAlgorithmProviders = [];
+            ConcurrentBag<ISignatureAlgorithmProvider> signatureAlgorithmProviders = [];
```

Comment:
These tests would be simpler without parallelism.  I think parallelism adds unnecessary complexity.  (Hence, the bug.)

See https://github.com/dotnet/sign/blob/dc00170e656abc2c5f17331a22196de096a2bb53/test/Sign.Core.Test/FileSystem/FileInfoComparerTests.cs#L12-L18.

Reply:
I added parallelism to also test the lock inside those methods. Otherwise I would do it as you suggested.